File: HRMS/Models/TaskCommentInsertUpdate.cs
using System.ComponentModel.DataAnnotations;

namespace HRMS.Models;

public class TaskCommentInsertUpdate
{
    [Required]
    public DateTime Time { get; set; }

    [Required]
    public string Content { get; set; } = string.Empty;

    [Required]
    public int? TaskId { get; set; }

    [Required]
    public int? EmployeeId { get; set; }
}

File: HRMS.Desktop/lib/models/requests/task_comment_insert_update.dart
import 'package:json_annotation/json_annotation.dart';

part 'task_comment_insert_update.g.dart';

@JsonSerializable()
class TaskCommentInsertUpdate {
  DateTime time;
  String content;  
  String? taskId;
  String? employeeId;

  TaskCommentInsertUpdate(
    this.time,
    this.content,
    this.taskId,
    this.employeeId,
  );

  factory TaskCommentInsertUpdate.fromJson(Map<String, dynamic> json) =>
      _$TaskCommentInsertUpdateFromJson(json);

  Map<String, dynamic> toJson() => _$TaskCommentInsertUpdateToJson(this);
}
